lib/customers.js

```diff
@@ -138,6 +138,154 @@ async function updateCustomer (id, data, userToken) {
   return getCustomerById(id)
 }
 
+/**
+ * Update all customer record
+ *
+ * @name save
+ * @function
+ *
+ * @param {string} id Customer's id
+ * @param {object} data Fields to update
+ *
+ * @returns {Promise} Newly updated Customer
+ */
+
+function edit (id, data, userToken) {
+  const defaults = [
+    'front_camera',
+    'id_card_data',
+    'id_card_photo',
+    'us_ssn',
+    'subscriber_info',
+    'name'
+  ]
+  const filteredData = _.pick(defaults, _.mapKeys(_.snakeCase, _.omitBy(_.isNil, data)))
+  if (_.isEmpty(filteredData)) return getCustomerById(id)
+  const formattedData = enhanceEditedPhotos(enhanceEditedFields(filteredData, userToken))
+  const defaultDbData = {
+    customer_id: id,
+    created: new Date(),
+    ...formattedData
+  }
+
+  const cs = new Pgp.helpers.ColumnSet(_.keys(defaultDbData),
+    { table: 'edited_customer_data' })
+  const onConflict = ' ON CONFLICT (customer_id) DO UPDATE SET ' +
+    cs.assignColumns({ from: 'EXCLUDED', skip: ['customer_id', 'created'] })
+  const upsert = Pgp.helpers.insert(defaultDbData, cs) + onConflict
+  return db.none(upsert)
+    .then(getCustomerById(id))
+}
+
+/**
+ * Add *edited_by and *edited_at fields with acting user's token
+ * and date of override respectively before saving to db.
+ *
+ * @name enhanceEditedFields
+ * @function
+ *
+ * @param {object} fields Fields to be enhanced
+ * @param {string} userToken Acting user's token
+ * @returns {object} fields enhanced with *_by and *_at fields
+ */
+
+function enhanceEditedFields (fields, userToken) {
+  if (!userToken) return fields
+  _.mapKeys((field) => {
+    fields[field + '_by'] = userToken
+    fields[field + '_at'] = 'now()^'
+  }, fields)
+  return fields
+}
+
+/**
+ * Add *_path to edited photos fields
+ *
+ * @name enhanceEditedFields
+ * @function
+ *
+ * @param {object} fields Fields to be enhanced
+ * @returns {object} fields enhanced with *_path
+ */
+
+function enhanceEditedPhotos (fields) {
+  return _.mapKeys((field) => {
+    if (_.includes(field, ['front_camera', 'id_card_photo'])) {
+      return field + '_path'
+    }
+    return field
+  }, fields)
+}
+
+/**
+ * Remove the edited data from the db record
+ *
+ * @name enhanceOverrideFields
+ * @function
+ *
+ * @param {string} id Customer's id
+ * @param {object} data Fields to be deleted
+ *
+ * @returns {Promise} Newly updated Customer
+ *
+ */
+
+function deleteEditedData (id, data) {
+  // NOT IMPLEMENTING THIS FEATURE FOR THE CURRENT VERSION
```

Comment:
Best to add a `TODO:` tag on this comment for easier identification later

```suggestion
  // TODO: NOT IMPLEMENTING THIS FEATURE FOR THE CURRENT VERSION
```